### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by tanya-newStart**
A meals router and a reservations router.

### CODE REVIEW

File: api/src/routers/meals.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  if (data.price !== undefined) {
+    const price = Number(data.price);
+    if (typeof price !== "number" || price <= 0) {
+      errors.push("Price must be a positive number.");
+    }
+  }
+
+  return errors;
```
Comment (magdazelena) [inline api/src/routers/meals.js:64]: very nice error handling and parsing!

File: api/src/routers/meals.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      title,
+      description,
+      location,
+      meal_when,
+      max_reservations,
+      price,
+      created_date,
+    } = req.body;
+
+    const updateData = {};
```
Comment (magdazelena) [inline api/src/routers/meals.js:125]: I generally recommend as a practice to name "active" elements such as functions using "active" names like `update` and other way around for passive, such as objects fx. `toUpdate` or `updated`. In this case, the object is collecting data to be updated, right? I would suggest `dataToUpdate` or something similar :)

**Review by magdazelena — APPROVED** [review]
very nice data validation and error handling! I added a small comment to the naming, but other than that, way to go!

File: api/src/routers/meals.js (hunk 2 above)
Comment (tanya-newStart) [reply api/src/routers/meals.js:125]: it does sound better! Thank you!

**Pull request merged**